conans/client/tools/files.py

```diff
@@ -363,3 +363,18 @@ def unix2dos(filepath):
 
 def dos2unix(filepath):
     _replace_with_separator(filepath, "\n")
+
+def rename(src, dst):
+    """
+    rename a file or folder to avoid "Access is denied" error on Windows
+    :param src: Source file or folder
+    :param dst: Destination file or folder
+    """
+    if platform.system() == "Windows" and which("robocopy") and os.path.isdir(src):
+        if os.path.exists(dst):
+            raise ConanException("rename {} to {} failed, dst exists.".format(src, dst))
+        retcode = os.system("robocopy /move /e /ndl /nfl {} {}".format(src, dst))
```

Comment:
If source or destination has spaces the command might fail, or?
=> Add quotes for src and dst

Reply:
Thanks for pointing out. As @uilianries said, I will replace os.system with subprocess which can handle the spaces in path more gracefully.

Reply:
> If source or destination has spaces the command might fail, or?
> => Add quotes for src and dst
Use subprocess and command sequence instead os.system(). I tested and it works for source or destination has spaces.

Reply:
Unit tests should cover the spaces thing (create temp folders, or use the TestClient, which automatically adds spaces in the path).